jax/_src/pallas/mosaic/primitives.py

```diff
@@ -63,7 +63,7 @@ def _repeat(x):
 bitcast_p = jax_core.Primitive("bitcast")
 
 
-def bitcast(x, ty: DTypeLike):
+def bitcast(x: ArrayLike, ty: DTypeLike) -> jax.Array:
```

Comment:
Seems like mypy doesn't like this annotation since we use `.shape`. Perhaps `Array` is a better fit here? Or we need to use `jnp.asarray` in the function. cc: @jakevdp

Reply:
Changed to `Array`.